src/browser/provider/pool.js

```diff
@@ -0,0 +1,145 @@
+            providerName = providerRegExpMatch[1];
+            browserName  = providerRegExpMatch[2] || '';
+        }
+        else {
+            providerName = 'local';
+            provider     = await this.getProvider(providerName);
+            browserName  = providerRegExpMatch[1] || '';
+        }
+
+        return { provider, providerName, browserName };
+    },
+
+    async _parseAlias (alias) {
+        if (alias && alias.path)
+            return this._handlePathAndCmd(alias);
+
+        if (typeof alias === 'string')
+            return this._parseAliasString(alias);
+
+        throw new GeneralError(MESSAGE.cantFindBrowser, alias);
+    },
+
+    async _getInfoForAllBrowserNames (provider, providerName) {
+        var allBrowserNames = await provider.listAvailableBrowserNames();
+
+        if (!allBrowserNames.length)
+            return { provider, providerName, browserName: '' };
+
+        return allBrowserNames
+            .map(browserName => ({ provider, providerName, browserName }));
+    },
+
+    _getProviderModule (providerName) {
+        try {
+            var providerObject = require(`testcafe-provider-${providerName}`);
+
+            this.addProvider(providerName, providerObject);
+            return this._getProviderFromCache(providerName);
+        }
+        catch (e) {
+            return null;
+        }
+    },
+
+    _getProviderFromCache (providerName) {
+        return this.providersCache[providerName] && this.providersCache[providerName].provider || null;
+    },
+
+    _getBuiltinProvider (providerName) {
+        var providerObject = BUILT_IN_PROVIDERS[providerName];
+
+        if (!providerObject)
+            return null;
+
+        this.addProvider(providerName, providerObject);
+
+        return this._getProviderFromCache(providerName);
+    },
+
+    async getBrowserInfo (alias) {
+        if (alias instanceof BrowserConnection)
+            return alias;
+
+        var browserInfo = await this._parseAlias(alias);
+
+        var { provider, providerName, browserName } = browserInfo;
+
+        if (browserName === 'all')
+            return await this._getInfoForAllBrowserNames(provider, providerName);
+
+        if (!await provider.isValidBrowserName(browserName))
+            throw new GeneralError(MESSAGE.cantFindBrowser, alias);
+
+        return browserInfo;
+    },
+
+    addProvider (providerName, providerObject) {
+        this.providersCache[providerName] = new BrowserProviderCacheItem(
+            new BrowserProviderPluginHost(providerObject, providerName)
+        );
+    },
+
+    removeProvider (providerName) {
+        delete this.providersCache[providerName];
+    },
+
+    getProviderSync (providerName) {
+        var provider = this._getProviderFromCache(providerName);
+
+        if (!provider)
+            provider = this._getProviderModule(providerName);
+
+        if (!provider)
+            provider = this._getBuiltinProvider(providerName);
+
+        return provider;
```

Comment:
Let's get rid of this method. Let's always use async method and initialize provider. We can make `createBrowserProvider` async in API